### ISSUE TITLE
Maybe there is some error in learner.py?

### ISSUE THREAD

**Issue by crashmoon** [dragen1860/MAML-Pytorch#29]
The author tries to define the running state of the BatchNorm manually. However, I found there is some error in learner.py, in line 66, and line 67.
###### learner.py ########
running_mean = nn.Parameter(torch.zeros(param[0]), requires_grad=False) # line 66 #
running_var = nn.Parameter(torch.ones(param[0]), requires_grad=False)     # line 67 #
#######################
The running_mean and the running_var in BatchNorm is not the parameter of the network itself, since it dose not require calculating gradient.
It is the running_stats of the model and only records the mean and var in each minibatch during iterations.
We can change line 66 and line 67 as following to avoid calculating gradient:
###############
self.register_buffer('running_mean',  XXXXXXXX)
self.register_buffer('running_var', XXXXXXXXXX)
###############
reference：
https://pytorch.org/docs/stable/_modules/torch/nn/modules/batchnorm.html#BatchNorm2d

**Comment by dragen1860** [comment]
Hi, running mean and variance do not required gradient. So I register it as non-gradient parameters.
I think its almost the same compared with register_buffer.
